### PR TITLE
Gitlab integration

### PR DESCRIPTION
**Pull request by jyothishnt**
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-2599

Integrated

- tests
- linting using flake8
- formatting using black

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Stages for the CI/CD job
 stages:
   - setup
+  - lint
+  - format
+  - test
```
Comment (azangru) [inline .gitlab-ci.yml:6]: Should `lint` and `format` really be independent stages / jobs? Considering this is CI/CD environment, and no changes to the source code are going to be committed, is there any reason to run the formatter at all? And doesn't it make sense to run the linter as part of the test job?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -79,6 +82,39 @@ Setup:Review:
...
+  before_script:
+    - pip install -r requirements.txt
+  script:
+    - flake8 .
+  only:
+    - branches
+    - merge_requests
+
+# Formatting
+Format:
```
Comment (jonk-ebi) [inline .gitlab-ci.yml]: Without a push command this stage will correct the code in the runner but nothing else.

Personally I don't think it is a good idea to run code modification stages in CI. 
My reasons are :-
- The author ends up with a slightly different version of the code to the PR
- Can introduce hard to discover errors 

Black should be ran as a precommit hook or on save in your IDE.

File: app/core/config.py [hunk 3]
```diff
@@ -29,7 +29,11 @@
...
 config = Config(".env")
 
-ENSEMBL_SEARCH_HUB_API: str = config("ENSEMBL_SEARCH_HUB_API", cast=str, default="http://ensembl-search-hub-svc:8083/api/search/stable-id")
+ENSEMBL_SEARCH_HUB_API: str = config(
+    "ENSEMBL_SEARCH_HUB_API",
+    cast=str,
+    default="http://ensembl-search-hub-svc:8083/api/search/stable-id",
+)
 DEFAULT_APP = config("DEFAULT_APP", cast=str, default="entity-viewer")
 ENSEMBL_URL = config("ENSEMBL_URL", cast=str, default="http://beta.ensembl.org")
```
Comment (kamaldodiya) [inline app/core/config.py:38]: https ?

File: requirements.txt [hunk 4]
```diff
@@ -3,3 +3,6 @@ fastapi==0.111.0
 loguru==0.7.2
 pydantic==2.7.1
 Requests==2.32.2
+flake8==7.1.1
+pre-commit=3.8.0
+black=24.8.0
```
Comment (kamaldodiya) [inline requirements.txt:8]: Should this be in requirements-dev.txt I don't think this is needed in production for example https://github.com/Ensembl/ensembl-web-tools-api/blob/dev/requirements-dev.txt

**Review by kamaldodiya — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 5]
```diff
@@ -3,3 +3,4 @@ fastapi==0.111.0
 loguru==0.7.2
 pydantic==2.7.1
 Requests==2.32.2
+flake8==7.1.1
```
Comment (kamaldodiya) [inline requirements.txt:6]: Is this still needed in production build ?

**Review by kamaldodiya — APPROVED** [review]

**Review by azangru — APPROVED** [review]

**Pull request merged**